File: packages/types/src/contributor.ts
import { Entity } from './index';

export type ContributorStatus = 'ACTIVE' | 'INACTIVE';

export interface Contributor {
  id: string;
  tenantId?: string;
  code: string;
  name: string;
  status: ContributorStatus;
  entityId?: string | null;
  entity?: Entity | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateContributorDto {
  code: string;
  name: string;
  status?: ContributorStatus;
  entityId?: string | null;
}

export interface UpdateContributorDto {
  code?: string;
  name?: string;
  status?: ContributorStatus;
  entityId?: string | null;
}

export interface ContributorWithStats extends Contributor {
  articlesCount: number;
}

export interface DuplicateContributorGroup {
  normalizedName: string;
  contributors: ContributorWithStats[];
  matchScore: number;
}

export interface MergeContributorsDto {
  primaryContributorId: string;
  secondaryContributorId: string;
}

export interface ContributorFilterQuery {
  search?: string;
  status?: ContributorStatus | 'ALL';
  page?: number;
  limit?: number;
}

File: apps/android/app/src/main/java/com/mmedic/data/model/Contributor.kt
package com.mmedic.data.model

data class Contributor(
    val id: String,
    val code: String,
    val name: String,
    val status: String = "ACTIVE",
    val entityId: String? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null
)

data class CreateContributorDto(
    val code: String,
    val name: String,
    val status: String = "ACTIVE",
    val entityId: String? = null
)

data class UpdateContributorDto(
    val code: String? = null,
    val name: String? = null,
    val status: String? = null,
    val entityId: String? = null
)

data class ContributorWithStats(
    val id: String,
    val code: String,
    val name: String,
    val status: String = "ACTIVE",
    val entityId: String? = null,
    val articlesCount: Int = 0,
    val createdAt: String? = null,
    val updatedAt: String? = null
)

data class DuplicateContributorGroup(
    val normalizedName: String,
    val contributors: List<ContributorWithStats>,
    val matchScore: Int = 95
)

data class MergeContributorsDto(
    val primaryContributorId: String,
    val secondaryContributorId: String
)
